app/native/IRContextMenuManager/IRContextMenuManager.mm

```diff
@@ -0,0 +1,174 @@
+#import "IRContextMenuManager.h"
+#import <Cocoa/Cocoa.h>
+#import <React/RCTUtils.h>
+
+static NSString * const kSeparatorString = @"menu-item-separator";
+
+@implementation IRContextMenuManager
+
+RCT_EXPORT_MODULE()
+
+- (std::shared_ptr<facebook::react::TurboModule>)getTurboModule:(const facebook::react::ObjCTurboModule::InitParams &)params {
+  return std::make_shared<facebook::react::NativeIRContextMenuManagerSpecJSI>(params);
+}
+
+#pragma mark - API
+
+- (void)showContextMenu:(NSArray *)items {
+  RCTExecuteOnMainQueue(^{
+    [self presentContextMenuWithItems:items];
+  });
+}
+
+#pragma mark - Helpers
+
+- (void)presentContextMenuWithItems:(NSArray *)items {
+  NSMenu *menu = [[NSMenu alloc] initWithTitle:@""];
+
+  [self buildMenu:menu fromEntries:items path:@[]];
+
+  NSEvent *event = [NSEvent mouseEventWithType:NSEventTypeRightMouseDown
+                                       location:[self currentMouseLocation]
+                                  modifierFlags:0
+                                      timestamp:[[NSProcessInfo processInfo] systemUptime]
+                                   windowNumber:[[NSApp keyWindow] windowNumber]
+                                        context:nil
+                                    eventNumber:0
+                                     clickCount:1
+                                       pressure:1.0];
```

Comment:
This is the typical way people write Objective-C, but I've been using this style instead, which I think just works better with my JavaScript brain.

```suggestion
  NSEvent *event = [NSEvent mouseEventWithType:NSEventTypeRightMouseDown
    location:[self currentMouseLocation]
    modifierFlags:0
    timestamp:[[NSProcessInfo processInfo] systemUptime]
    windowNumber:[[NSApp keyWindow] windowNumber]
    context:nil
    eventNumber:0
    clickCount:1
    pressure:1.0
  ];
```